packages/replay/src/replay.ts

```diff
@@ -1151,6 +1151,12 @@ export class ReplayContainer implements ReplayContainerInterface {
       return;
     }
 
+    const eventBuffer = this.eventBuffer;
+    if (eventBuffer && this.session.segmentId === 0 && !eventBuffer.hasCheckout) {
```

Comment:
Hmm won't this log for session based replays?

Reply:
there should still always be a checkout in the initial segment, right?